### PR TITLE
Use `pytorch-lightning>=1.5.0`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Use recent PyTorch-lightning, `1.5.*` to resolve https://github.com/optuna/optuna/pull/3077.

## Description of the changes

- Fix the error by following https://github.com/optuna/optuna/pull/3077#pullrequestreview-796783965
- Update the PyTorch-lightning version in `setup.py` because the callback implementation does not work with `pytorch-lightning<1.5` anymore.
- Fix warning in the test: replace `checkpoint_callback` with `enable_checkpointing`

### CODE REVIEW

**Comment by hvy** [comment]
Thanks for the fix!

@toshihikoyanase , @himkt could you review this one?

File: optuna/integration/pytorch_lightning.py [hunk 1]
```diff
@@ -56,10 +56,10 @@ def __init__(self, trial: optuna.trial.Trial, monitor: str) -> None:
         self.is_ddp_backend = False
 
     def on_init_start(self, trainer: Trainer) -> None:
-        self.is_ddp_backend = trainer.accelerator_connector.distributed_backend is not None
+        self.is_ddp_backend = trainer._accelerator_connector.distributed_backend is not None
```
Comment (toshihikoyanase) [inline optuna/integration/pytorch_lightning.py:59]: [Question] Does pytorch-lightning provide any public interface to check the distributed backend?

Comment (nzw0301) [reply optuna/integration/pytorch_lightning.py:59]: I'm afraid that I was not able to find such an interface on the [trainer page](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.trainer.trainer.html?highlight=trainer#trainer).

**Review by himkt — APPROVED** [review]

File: optuna/integration/pytorch_lightning.py (hunk 1 above)
Comment (toshihikoyanase) [reply optuna/integration/pytorch_lightning.py:59]: That makes sense. Thank you for your investigation.

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you for your fix.

**Pull request merged**